src/lib.rs

```diff
@@ -179,33 +195,53 @@ impl Profile {
         decoder: &mut Decoder,
     ) {
         let pc = machine.pc().to_u64();
+        let sp = machine.registers()[SP].to_u64();
+        if sp < self.sbrk_heap {
+            panic!("Heap and stack overlapping sp={} heap={}", sp, self.sbrk_heap);
+        }
         let inst = decoder.decode(machine.memory_mut(), pc).unwrap();
         let opcode = ckb_vm::instructions::extract_opcode(inst);
         let cycles = machine.instruction_cycle_func().as_ref().map(|f| f(inst)).unwrap_or(0);
         self.trie_node.borrow_mut().cycles += cycles;
         self.trie_node.borrow_mut().pc = pc;
 
         let call = |s: &mut Self, addr: u64, link: u64| {
+            let mut regs = [[0; 32]; 2];
+            for i in 0..32 {
+                regs[0][i] = machine.registers()[i].to_u64();
```

Comment:
Can we save registers which are actually used?

Reply:
I think I might use them in the future (though not now)